Explicate what "Rc" and "Arc" stand for.

ucarion opened the pull request:
A person on the weekly "Easy Questions" Reddit thread [was mystified by what `Arc`/`Rc` means](https://www.reddit.com/r/rust/comments/6dyud9/hey_rustaceans_got_an_easy_question_ask_here/did87ds/). Though this is explained in various places, it's not mentioned in the documentation directly.

This PR adds an explanation of the `Rc`/`Arc` acronyms to their respective documentations. There are two things I'm not sure of:

* Does "Rc" mean "Reference Count**er**" or "Reference Count**ed**"? ~~I went with the former.~~ *Edit:* I've changed this to use the latter alternative.
* Should this information be spelled out elsewhere, such as in the docs for the `rc` module?

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @sfackler (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

sfackler commented on src/liballoc/rc.rs:
```diff
@@ -10,7 +10,8 @@
 #![allow(deprecated)]
 
-//! Single-threaded reference-counting pointers.
+//! Single-threaded reference-counting pointers. "Rc" stands for "Reference
```
I think of Rc as "reference counted", not "reference counter" personally, and Arc as "atomically reference counted".

ucarion replied:
After reflection, I was reaching the same conclusion. I'll fix the PR.

nodakai replied:
When I compare "atomically reference-counted" with "atomic reference counter," the tech-jargon-ness of the former seems a bit stronger than that of the latter. Just my personal impression.

sfackler commented:
@rust-lang/docs

abrugh commented:
Hey There. Thanks @ucarion for creating a change request for me. Just commenting to add context. [Here's](https://www.reddit.com/r/rust/comments/6dyud9/hey_rustaceans_got_an_easy_question_ask_here/difmp6f/?context=3 ) the comment in question from reddit.

steveklabnik commented:
> Does "Rc" mean "Reference Counter" or "Reference Counted"? I went with the former. Edit: I've changed this to use the latter alternative.

Yes, the latter is correct.

> Should this information be spelled out elsewhere, such as in the docs for the rc module?

That might be a good idea, yes 😄

steveklabnik commented on src/liballoc/arc.rs:
```diff
@@ -42,7 +42,8 @@ use heap::deallocate;
 const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 
-/// A thread-safe reference-counting pointer.
+/// A thread-safe reference-counting pointer. "Arc" stands for "Atomically
```
Both of these should be single quotes

ucarion commented:
>>Should this information be spelled out elsewhere, such as in the docs for the rc module?

>That might be a good idea, yes 😄

@steveklabnik I misspoke -- I had added those docs to the `rc` module all along. With the latest commit, I also added docs to the `Rc` struct as well. (I also switched to single quotes)

frewsxcv commented:
This looks great, thanks!

@bors r+ rollup

bors commented:
:pushpin: Commit 1af0cb1 has been approved by `frewsxcv`

<!-- @bors r=frewsxcv 1af0cb165065ebca5296a7b5841ebec9bfb40476 -->

bors commented:
:hourglass: Testing commit 1af0cb1 with merge 54eeef1...

bors commented:
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.3552), [status-travis](https://travis-ci.org/rust-lang/rust/builds/241887264?utm_source=github_status&utm_medium=notification)
Approved by: frewsxcv
Pushing 54eeef14a380ca026dc378e6927655fb08f64736 to master...
